### PR TITLE
[RDY] Add soccer items

### PR DESCRIPTION
**Pull request by BrettDong**
### Summary of this pull request
- [x] New items: soccer, soccer jerseys, soccer shorts, soccer shinguards, soccer socks, goalkeeper gloves
- [X] New professions: soccer player, soccer goalkeeper
- [X] Splited football jerseys, soccer jerseys and other sports apart into different ids. This prevents baseball jerseys from appearing in a football field.
### A brief list of things to be implemented in future PRs
- Soccer stadium
- Press [a] to kick soccer
- Ability of catch-and-throw-back or directing away incoming flying objects of a soccer goalkeeper

### CODE REVIEW

**Comment by Coolthulhu** [comment]
> New action: Kick something

Kicking sounds like a rather complex addition that shouldn't be added to a new button, but integrated with others.

For example, making kicking a martial art style with a special flag that makes it not use wielded weapon but just kicks.

I specifically do NOT want any of those to happen:
* Having to use kick->direction to attack if your kicking ability gets better than punching ability or if your current weapon sucks (for example, you carry a gas tank)
* Having to drop an item and then kick it rather than just using throw. You can just special case it in the throw function and give some bonuses and flavor message
* Having to remember a new keybind to kick open doors. Just special case it in smash.

> Different strength or skill for different body parts

We already have problems with finding enough different roles for 4 stats. If they multiplied, it would be even worse.

**Comment by Coolthulhu** [comment]
Any stat-boosting ability means the profession goes into the "more professions" mods.

**Comment by BrettDong** [comment]
What if the profession costs the same points the stats it gains?

**Comment by Coolthulhu** [comment]
Then it still allows using skill points to boost stats.
We don't have a mechanism of having profession cost stat points yet.

**Comment by BrettDong** [comment]
Oh yeah, I forgot the points of scenarios, professionals and stats are independent.

**Comment by Cyrano7** [comment]
Is there a way to make any of this more generic? Like soccer shorts -> athletic shorts, soccer socks -> knee-high socks or soccer shin guards -> just shin guards. Adding such specific items is kinda cluttery when playing a non soccer specific build.

**Comment by BrettDong** [comment]
Maybe I'll add some special use of them in the future, so I created that particular items.

File: data/json/items/armor.json [hunk 1]
```diff
@@ -1227,6 +1227,27 @@
     "flags": [ "VARSIZE", "OVERSIZE", "STURDY" ]
   },
   {
+    "id": "goalkeeper_gloves",
+    "type": "ARMOR",
+    "name": "pair of goalkeeper gloves",
+    "name_plural": "pairs of goalkeeper gloves",
+    "description": "A pair of professional green goalkeeper gloves.  Helped the user hit a ball flying directly at a hundred mile per hour without injury before.  Needless to say how much protection it offers.",
```
Comment (illi-kun) [inline data/json/items/armor.json]: ```a hundred miles per hour``` (mile -> miles).

File: data/json/items/armor.json [hunk 2]
```diff
@@ -5084,6 +5135,92 @@
...
+      }
+    ],
+    "flags": [ "VARSIZE" ]
+  },
+  {
+    "id": "soccer_shorts",
+    "type": "ARMOR",
+    "name": "soccer shorts",
+    "name_plural": "soccer shorts",
+    "description": "A pair of loose-fitting soccer shorts, with a logo of a soccer team and a nubmer.",
```
Comment (illi-kun) [inline data/json/items/armor.json]: Typo in 'nubmer'

**Comment by mugling** [comment]
> Adding such specific items is kinda cluttery when playing a non soccer specific build.

I'd agree with both this and @Coolthulhu's comment. This all needs carefuly working in to the game as novel mechanics otherwise it's just item bloat. Alternatively you could move this to more professions mod.

**Comment by Oskar636** [comment]
Killing zombies with football kicking seems ridiculous, add the fact that DDA has loads of diffrent, more dangerous zombies, I don't know how this would make sense. I like the catch/throw thing, but it still doesn't seem to useful ( you don't get thrown at too often in this game ). Stadiums are a strange option in my opinion because I don't know what loot you'd find there and I'd imagine loads of zombies.

**Pull request closed without merging**